File: frontend/angular-ecommerce/src/app/commons/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageUrl:string='';
    quantity:number=0;
    unitPrice:number=0;
    productId:number=0;

    constructor(cartItem:CartItem) { 
        this.imageUrl = cartItem.imageUrl;
        this.quantity = cartItem.quantity;
        this.unitPrice = cartItem.unitPrice;
        this.productId = cartItem.id;
    }
}

File: backend/spring-boot-ecommerce/src/main/java/com/example/ecommerce/entity/OrderItem.java
package com.example.ecommerce.entity;

import java.math.BigDecimal;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Entity
@Table(name = "order_item")
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "unit_price")
    private BigDecimal unitPrice;

    @Column(name = "quantity")
    private int quantity;


    @Column(name = "product_id")
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
}
